Update Sagemaker Guide (#1534)
astronomer/docs: learn/airflow-sagemaker.md

## Step 1: Create a role to access SageMaker

For this tutorial, you will need to access SageMaker from your Airflow environment.  There are multiple ways to do this, but for this tutorial you will create an AWS role that can access SageMaker and create temporary credentials for that role.

:::info

> If you are uncertain which method you use to connect to AWS, contact your AWS Administrator. These steps may not fit your desired authentication mechanism

Reviewer: ```suggestion
If you are uncertain which method you use to connect to AWS, contact your AWS Administrator. These steps may not fit your desired authentication mechanism.
```

    These variables ensure that all SageMaker operators will work in your Airflow environment. Some require XCom pickling to be turned on in order to work because they return objects that are not JSON serializable. 

>     Your `.env` file only applies to your local Airflow. To set variables in your Astronomer Airflow instances, [follow this guide](https://docs.astronomer.io/astro/environment-variables).

Reviewer: This tutorial is explicitly written for users running locally with the CLI (hence step 2 on setting up a local Airflow project). If this is causing confusion for Astro customers we can add it as a separate note, but I don't want to put it directly in the step.

Author: ```suggestion
```

## Step 4: Add Airflow Variables

> Add an Airflow variable that will be used by your DAG. In the Airflow UI, go to **Admin** -> **Variables**.

Reviewer: I think @fhoda specifically updated this DAG to store the S3 bucket as an env var. Is there a specific reason you're adding it in the DAG code instead?

Reviewer 2: For some context, I did this so that users know what exactly they need to change by just adding the vars rather than having to go through the code to figure out what they need to change to get it working in their environments. It may be good to call out that in production it may be considered best practice to hard code the s3 bucket and similar things so that they can be version controlled.

Author: That makes sense. I partially changed it because I didn't want to deal with f-strings and jinja templating :)

Author: Would that be a good trend in general - things users definitely need to change in these tutorials should be airflow variables?

}
```

:::warning

> Your credentials should only last up to a day - you will need to edit the details in the connection with new credentials, if you want to run this again another day

Reviewer: ```suggestion
Your AWS credentials will only last one day using this authentication method. If you return to this step after your credentials have expired, you will need to edit the details in the connection with updates credentials.
```

instance) and `role` (Role ARN to execute SageMaker jobs) then fill in the information directly below with the target
AWS S3 locations, and model and training job names.
"""
import os

# Define variables used in configs
data_url = "https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data"  # URL for Iris data API
date = "{{ ts_nodash }}"  # Date for transform job name

bucket = "my_bucket"
input_s3_key = 'iris/processed-input-data'  # Train and test data S3 path
output_s3_key = 'iris/results'  # S3 path for output data
model_name = "Iris-KNN-{}".format(date)  # Name of model to create
training_job_name = 'train-iris-{}'.format(date)  # Name of training job
model_name = f"Iris-KNN-{date}"  # Name of model to create
training_job_name = f'train-iris-{date}'  # Name of training job
region = "us-east-2"

with DAG('sagemaker_pipeline',
         start_date=datetime(2021, 7, 31),
         start_date=datetime(2022, 1, 1),
         max_active_runs=1,
         schedule_interval=None,
         default_args={
             'retries': 0,
             'retry_delay': timedelta(minutes=1),

>              'aws_conn_id': 'aws-sagemaker'

Reviewer: I'm confused why we would remove this from the default when it's used by every task?

Author: I think it's a better practice to be explicit - especially with something like `aws_conn_id`. I think new-to-airflow users may copy-paste one of the operators out into their own DAG and wouldn't think to look in `default_args` - so I always try to make it obvious and to get people in the habit of being explicit with their `*_conn_id`'s in operators as it can be a source of confusion
